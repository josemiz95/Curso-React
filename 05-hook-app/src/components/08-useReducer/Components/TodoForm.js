import React from 'react';
import { useForm } from '../../../hooks/useForm';

export const TodoForm = ({handleAddTodo}) => {
    const [{description}, inputChange,reset] = useForm({description:''});

    const handleSubmit = (e) =>{
        e.preventDefault();

        if(description.trim() === '') return;
        
        handleAddTodo(description);
        reset();
    }

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" name="description" placeholder="Tarea" autoComplete="off" className='form-control'
                value={description} onChange={inputChange} />
            <button type='submit' className='btn btn-outline-success mt-1 btn-block'>Agregar</button>
        </form>
    )
}
