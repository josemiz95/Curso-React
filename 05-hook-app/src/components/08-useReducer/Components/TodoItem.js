import React from 'react'

export const TodoItem = ({todo, handleDelete, handleToggle}) => {
    return (
        <li className="list-group-item">
            <p className={`${todo.done && 'complete'}`} onClick={() => {handleToggle(todo.id)}}>{todo.desc}</p>
            <button className='btn btn-danger' onClick={()=> {handleDelete(todo.id)}}>Borrar</button>
        </li> 
    )
}
