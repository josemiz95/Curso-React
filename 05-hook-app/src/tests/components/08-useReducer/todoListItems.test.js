import { shallow } from "enzyme";
import { TodoItem } from "../../../components/08-useReducer/Components/TodoItem";
import { demoTodos } from "../../fixtures/demoTodos";

describe('Test for <TodoItem/>', () => {
	const todo = demoTodos[0];
	const handleDelete = jest.fn();
	const handleToggle = jest.fn();
	const wrapper = shallow(<TodoItem todo={todo} handleDelete={handleDelete} handleToggle={handleToggle} />);

	test('should show correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should trigger handleDelete', () => {
		wrapper.find('button').simulate('click');
		expect(handleDelete).toHaveBeenCalledWith(todo.id);
	});

	test('should trigger toggle', () => {
		wrapper.find('p').simulate('click');
		expect(handleToggle).toHaveBeenCalledWith(todo.id);
	});

	test('text should be show correctly', () => {
		const text = wrapper.find('p').text();
		expect(text).toBe(todo.desc);
	});

	test('should have the correct class when done', () => {
		todo.done = true;
		const wrapper = shallow(<TodoItem todo={todo}/>);
		expect(wrapper.find('p').hasClass('complete')).toBeTruthy();
	});
});