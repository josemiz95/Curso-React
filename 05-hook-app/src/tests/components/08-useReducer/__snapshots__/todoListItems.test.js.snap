// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <TodoItem/> should show correctly 1`] = `
<li
  className="list-group-item"
>
  <p
    className="false"
    onClick={[Function]}
  >
    Algo
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
